raiden_contracts/tests/fixtures/channel.py

```diff
@@ -752,4 +758,8 @@ def call_settle(token_network, channel_identifier, A, vals_A, B, vals_B):
         vals_B.transferred,
         vals_B.locked,
         vals_B.locksroot,
-    ).transact({'from': A})
+    )
+    # call() raises TransactionFailed exception
+    contract_function.call({'from': A})
+    # transact() changes the chain state
+    contract_function.transact({'from': A})
```

Comment:
Why don't you use `call_and_transact` in this case?

Reply:
This is an unfortunate leftover.

How can I prevent this kind of mistakes?

Reply:
Often you can spot them in the diff before committing, but in this case the diff is quite large so it's hard to spot. I can't think of anything better than just "fix it when you see it", like it is the case for many refactoring opportunities.

Reply:
Alright.  I'm just worried about tests silently succeeding but with transactions failing.

Reply:
The heavy handed approach to this would be to forbid usage of `transact` in tests by replacing it with either
```
    raise AssertionError('transact must not be called in test code')
```
or with `call_and_transact`. If you do it with `pytest.monkeypatch` you can even undo it for a few cases where needed. But I'm not sure I want to recommend such an invasive change.

Reply:
The disabling of `transact()` is deferred to https://github.com/raiden-network/raiden-contracts/issues/821